repos/system_upgrade/el7toel8/actors/mariadbcheck/libraries/library.py

```diff
@@ -0,0 +1,192 @@
+            if option_name == changed_opt:
+                warn_changed.append(changed_opt)
+
+        for renamed_opt in options_renamed:
+            if option_name == renamed_opt:
+                warn_renamed[renamed_opt] = options_renamed[renamed_opt]
+
+        for removed_opt in options_not_in_103:
+            if option_name == removed_opt:
+                warn_removed.append(removed_opt)
+
+    return (warn_changed, warn_renamed, warn_removed)
+
+
+def generate_report():
+    """ Generate reports informing user about possible manual intervention requirements """
+
+    DOC_URL = 'https://access.redhat.com/articles/4055661'
+    MYCNF = '/etc/my.cnf'
+
+    config = parse_mycnf(MYCNF)
+    unsupported_plugins = get_unsupported_plugins(get_plugin_dir(config))
+    (opts_changed, opts_renamed, opts_removed) = get_warn_options(config)
+
+    # Show documentation url if mariadb-server installed
+    title = 'MariaDB server installation detected'
+    summary = 'MariaDB server will be upgraded. Additional steps might be required.\n\n'
+
+    reporting.create_report([
+        reporting.Title(title),
+        reporting.Summary(summary),
+        reporting.Severity(reporting.Severity.HIGH),
+        reporting.Tags([reporting.Tags.SERVICES]),
+        reporting.ExternalLink(title='Read more here.', url=DOC_URL)
+        ])
+
+    if unsupported_plugins:
+        title = 'MariaDB server unsupported plugins detected'
+        summary = 'Following installed plugins won\'t be part of MariaDB distribution ' \
+                  'after upgrade:\n{}'.format(", ".join(unsupported_plugins))
+
+        reporting.create_report([
+            reporting.Title(title),
+            reporting.Summary(summary),
+            reporting.Severity(reporting.Severity.HIGH),
+            reporting.Tags([reporting.Tags.SERVICES])
+            ])
+
+    if opts_changed:
+        title = 'MariaDB server changed default values of some options'
+        summary = 'Following config options changed default value:\n{}'.format(
+            ", ".join(opts_changed)
+        )
+
+        reporting.create_report([
+            reporting.Title(title),
+            reporting.Summary(summary),
+            reporting.Severity(reporting.Severity.MEDIUM),
+            reporting.Tags([reporting.Tags.SERVICES])
+            ])
+
+    if opts_renamed:
+        title = 'MariaDB server renamed some options'
+        severity = 'high'
+        summary = 'Following used config options were renamed:\n{}'.format(
+            ", ".join(opts_renamed)
+        )
+
+        reporting.create_report([
+            reporting.Title(title),
+            reporting.Summary(summary),
+            reporting.Severity(reporting.Severity.HIGH),
+            reporting.Tags([reporting.Tags.SERVICES])
+            ])
+
+    if opts_removed:
+        title = 'MariaDB server removed some options'
+        severity = 'high'
+        summary = 'Following used config options were removed:\n{}'.format(
```

Comment:
Should we still continue with the upgrade in this case? Or rather inhibit the upgrade?